src/mbgl/util/mapbox.cpp

```diff
@@ -121,7 +135,8 @@ std::string normalizeTileURL(const std::string& url, const std::string& accessTo
         return url;
     }
 
-    return baseURL + "v4/" + url.substr(sizeof("mapbox://tiles/") - 1) + "?access_token=" + accessToken;
+    auto query = normalizeQuery(url);
+    return baseURL + "v4/" + url.substr(sizeof("mapbox://tiles/") - 1, query.second - sizeof("mapbox://tiles/") + 1) + "?access_token=" + accessToken + normalizeQuery(url).first;
```

Comment:
If no query is present, `query.second` is `std::string::npos`, `query.second - sizeof("mapbox://tiles/") + 1` will be a huge number, and `url.substr(...)` will throw `std::out_of_range`. Please handle and add a test case for this.